R/import-check.R

```diff
@@ -75,3 +81,28 @@ import_check <- function(x, virtualenv_name) {
 
   out
 }
+
+env_type <- function(envname) {
+  ret <- NA
+  if(envname %in% virtualenv_list()) {
+    ret <- "virtualenv"
+  }
+  if(envname %in% conda_list()$name && is.na(ret)) {
```

Comment:
Rather than materializing a full list of all the conda environments on the system, I think it'd be better to resolve a specific envname. Something like


```r
env_python <- function(envname) { 
  if(virtualenv_exists(envname))
    return(virtualenv_python(envname))
  tryCatch(return(conda_python(envname)), 
           error = function(e) NULL)
}
  
```